### PR TITLE
fix: trim workspace and project name whitespace [DET-7390] [DET-7747]

### PR DESCRIPTION
**Pull request by EmilyBonar**
## Description
This PR trims whitespace from workspace and project names, as well as re-expanding long pinned workspace names after the navbar is collapsed and re-expanded [DET-7390] [DET-7747].

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DET-7390]: https://determinedai.atlassian.net/browse/DET-7390?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[DET-7747]: https://determinedai.atlassian.net/browse/DET-7747?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webui/react/src/hooks/useModal/Workspace/useModalWorkspaceCreate.tsx [hunk 1]
```diff
@@ -57,7 +57,7 @@ const useModalWorkspaceCreate = ({ onClose }: Props): ModalHooks => {
       closable: true,
       content: modalContent,
       icon: null,
-      okButtonProps: { disabled: name.length === 0 },
+      okButtonProps: { disabled: name.trim().length === 0 || name.trim().length > 80 },
```
Comment (hamidzr) [inline webui/react/src/hooks/useModal/Workspace/useModalWorkspaceCreate.tsx:60]: nit: share the logic for what a valid name is between workspace create and edit

File: webui/react/src/hooks/useModal/Project/useModalProjectCreate.tsx [hunk 2]
```diff
@@ -51,7 +51,11 @@ const useModalProjectCreate = ({ onClose, workspaceId }: Props): ModalHooks => {
 
   const handleOk = useCallback(async () => {
     try {
-      const response = await createProject({ description, name, workspaceId });
+      const response = await createProject({
+        description: description.trim(),
```
Comment (hamidzr) [inline webui/react/src/hooks/useModal/Project/useModalProjectCreate.tsx:55]: if we always want this, it could be something that `createProject` does.

Comment (hamidzr) [reply webui/react/src/hooks/useModal/Project/useModalProjectCreate.tsx:55]: or better probably a backend treatment but that might take longer

**Review by trentwatt — APPROVED** [review]
Looks fine, agree with @hamidzr about handling it a layer or two down.

Overall I woudn't see this as release blocking, because somebody has to actually go and type the trailing space, which seems unlikely.

But the sidebar fix we should definitely get in, and that looks good.

**Review by garrett361 — APPROVED** [review]
I also think Hamid's suggestions should be considered. Otherwise, it looks good to me.  

I verified that all leading and trailing white space is trimmed when created in the WebUI and that subsequent experiment submission to the trimmed Workspace/Project names is successful.

File: webui/react/src/hooks/useModal/Project/useModalProjectCreate.tsx (hunk 2 above)
Comment (garrett361) [reply webui/react/src/hooks/useModal/Project/useModalProjectCreate.tsx:55]: Backend would be nice. One can still create Workspaces/Projects with leading or trailing whitespace via the CLI, but you'd really have to go out of your way to do so.

**Pull request merged**